components/purchase_modal/purchase_modal.tsx

```diff
@@ -1072,6 +1166,57 @@ class PurchaseModal extends React.PureComponent<Props, State> {
         if (!stripePromise) {
             stripePromise = loadStripe(STRIPE_PUBLIC_KEY);
         }
+
+        if (this.props.isComplianceBlocked) {
+            return (
+                <Elements
```

Comment:
Nit: I think we can avoid wrapping this in Stripe's `<Elements>` provider for this branch. It will avoid some unnecessary work in the case that they come back to this later while still in screening. I think the loadStripe block up above can also be skipped if they are in screening.